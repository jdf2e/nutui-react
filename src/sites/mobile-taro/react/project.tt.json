{
  "miniprogramRoot": "./",
  "projectname": "first",
  "appid": "testAppId",
  "setting": {
    "es6": false,
    "minified": false
  }
}
