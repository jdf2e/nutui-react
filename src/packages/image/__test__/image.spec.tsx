import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Image from '@/packages/image'

const src =
  'https://img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg'

test('image props test', () => {
  const { container } = render(<Image src={src} width="100" height="100" />)
  expect(container.querySelector('.nut-image')).toHaveAttribute(
    'style',
    'height: 100px; width: 100px;'
  )
  expect(container.querySelector('.nut-img')).toHaveAttribute(
    'src',
    'https://img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg'
  )
  expect(container.querySelector('.nut-img')).toHaveAttribute(
    'style',
    'object-fit: fill; object-position: center;'
  )
  expect(container).toMatchSnapshot()
})

test('image position test', () => {
  const { container } = render(
    <Image src={src} width="100" height="100" fit="cover" position="left" />
  )
  expect(container.querySelector('.nut-img')).toHaveAttribute(
    'style',
    'object-fit: cover; object-position: left;'
  )
})

test('image round test', () => {
  const { container } = render(
    <Image src={src} width="100" height="100" radius="50%" />
  )
  expect(container.querySelector('.nut-image')).toHaveAttribute(
    'style',
    'height: 100px; width: 100px; overflow: hidden; border-radius: 50%;'
  )
})
