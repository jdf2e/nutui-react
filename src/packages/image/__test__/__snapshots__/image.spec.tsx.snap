// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image props test 1`] = `
<div>
  <div
    class="nut-image  "
    style="height: 100px; width: 100px;"
  >
    <img
      alt=""
      class="nut-img"
      src="https://img10.360buyimg.com/ling/jfs/t1/181258/24/10385/53029/60d04978Ef21f2d42/92baeb21f907cd24.jpg"
      style="object-fit: fill; object-position: center;"
    />
    <div
      class="nut-img-loading"
    >
      <svg
        aria-labelledby=""
        class="nut-icon"
        role="presentation"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M853.161 892.55H362.595l-209.433-.414a18.5 18.5 0 0 1-1.813-.092c-24.849-2.485-47.825-14.06-64.696-32.595-16.991-18.665-26.35-42.855-26.35-68.112V264.626c0-55.808 45.404-101.211 101.212-101.211H853.16c55.808 0 101.21 45.403 101.21 101.211v225.513c0 .275-.006.552-.018.826-.022.494-1.972 51.723 15.481 85.462 4.717 9.119 1.148 20.335-7.97 25.053-9.117 4.714-20.335 1.15-25.053-7.97-21.3-41.177-19.845-97.643-19.619-103.74V264.627c0-35.307-28.724-64.031-64.03-64.031H161.515c-35.308 0-64.032 28.724-64.032 64.031v526.711c0 32.755 24.321 59.958 56.718 63.62l208.431.412h490.53c35.306 0 64.03-28.725 64.03-64.032l-.382-93.677c-.105-1.75-1.587-19.548-19.55-42.5a9.548 9.548 0 0 1-.147-.19l-21.712-28.489-117.227-155.119c-.416-.55-38.995-50.852-86.272-45.534-38.335 4.315-75.955 45.164-108.79 118.132-17.615 39.141-34.65 68.269-52.082 89.046-17.607 20.986-35.68 33.52-55.251 38.317-43.423 10.638-81.05-18.995-120.887-50.373l-5.057-3.98c-46.555-36.578-68.75-28.224-158.33 59.602-7.33 7.188-19.1 7.071-26.289-.26-7.187-7.33-7.07-19.1.26-26.287 46.659-45.746 77.544-72.727 107.086-84.283 33.358-13.048 64.275-6.267 100.242 21.994l5.093 4.008c33.923 26.72 63.22 49.795 89.029 43.467 25.618-6.279 53.3-42.114 82.28-106.509 39.14-86.976 85.838-134.027 138.797-139.849 68.454-7.515 117.824 57.405 119.892 60.172l117.122 154.98 21.6 28.344c26.277 33.631 27.334 61.639 27.367 64.725l.001.128.384 94.06c.001 55.884-45.4 101.287-101.209 101.287z"
          fill="currentColor"
          fill-opacity="0.9"
        />
        ,
        <path
          d="M312.328 446.968c-42.325 0-76.759-34.434-76.759-76.76s34.434-76.759 76.76-76.759 76.759 34.435 76.759 76.76-34.434 76.759-76.76 76.759zm0-116.34c-21.824 0-39.579 17.756-39.579 39.58s17.755 39.58 39.58 39.58 39.579-17.756 39.579-39.58-17.755-39.58-39.58-39.58z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
  </div>
</div>
`;
